### PR TITLE
Support persistent class loader name tracking in all cases when JITServer AOT cache is active

### PR DESCRIPTION
**Pull request by cjjdespres**
The persistent class loader table will now track the name of the first-loaded class of particular class loaders whenever `getJITServerUseAOTCache()` is true, including when a local SCC is not present. If the ROM class of the first-loaded class happens to be shared, the table will save a pointer to its name in the local SCC. Otherwise, we create a persistent copy of the name.

A query function `lookupClassNameAssociatedWithClassLoader` has been added to look up only the class name associated to a class loader. Presently it is unused, but it will be useful in https://github.com/eclipse-openj9/openj9/pull/18301.

The `isROMClassInSharedCache` query function is now public. I was a little too ambitious making it private in https://github.com/eclipse-openj9/openj9/pull/18821 - it's still sometimes useful as a query, but only for lower-level ROM class processing when there is local SCC guaranteed to be present.

### CODE REVIEW

**Comment by cjjdespres** [comment]
Attn @mpirvu.

File: runtime/compiler/env/ClassLoaderTable.cpp [hunk 1]
```diff
@@ -182,28 +203,29 @@ TR_PersistentClassLoaderTable::associateClassLoaderWithClass(J9VMThread *vmThrea
...
-      if (useAOTCache && TR::Options::getVerboseOption(TR_VerboseJITServer))
-         TR_VerboseLog::writeLineLocked(TR_Vlog_JITServer,
-            "ERROR: Failed to get class chain for %.*s loaded by %p",
-            nameLength, (const char *)name, loader
-         );
-#endif /* defined(J9VM_OPT_JITSERVER) */
+   void *chain = _sharedCache ? _sharedCache->rememberClass(clazz) : NULL;
+   // If we are using the JITServer AOT cache, we need to remember the name of clazz with or without chain.
+   // If we aren't using it, there is no point in continuing.
+   if (!chain && !useAOTCache)
```
Comment (mpirvu) [inline runtime/compiler/env/ClassLoaderTable.cpp:210]: Ok, just a note than in the future we may not have an AOT cache or a SCC and still want to do this.

File: runtime/compiler/env/ClassLoaderTable.cpp [hunk 2]
```diff
@@ -182,28 +203,29 @@ TR_PersistentClassLoaderTable::associateClassLoaderWithClass(J9VMThread *vmThrea
...
+      auto romName = J9ROMCLASS_CLASSNAME(romClass);
+      nameStr = (_sharedCache && _sharedCache->isROMClassInSharedCache(romClass)) ? romName : copyJ9UTF8(romName, _persistentMemory);
+      name = J9UTF8_DATA(romName);
+      nameLength = J9UTF8_LENGTH(romName);
       }
-   TR_ASSERT(_sharedCache->isPointerInSharedCache(chain), "Class chain must be in SCC");
+#endif /* defined(J9VM_OPT_JITSERVER) */
 
-   info = new (_persistentMemory) TR_ClassLoaderInfo(loader, chain);
+   info = new (_persistentMemory) TR_ClassLoaderInfo(loader, chain, nameStr);
```
Comment (mpirvu) [inline runtime/compiler/env/ClassLoaderTable.cpp:228]: Do we delete such entries from the table? How do we handle the freeing of persistent memory for the name of the class? We much remember whether we used persistent memory, in which case we need to free it, or not, in which case we don't have to touch it.

File: runtime/compiler/runtime/JITServerAOTDeserializer.cpp [hunk 3]
```diff
@@ -388,6 +388,14 @@ JITServerAOTDeserializer::cacheRecord(const ClassLoaderSerializationRecord *reco
          );
       return false;
       }
+   else if (!chain)
+      {
+      if (TR::Options::getVerboseOption(TR_VerboseJITServer))
+         TR_VerboseLog::writeLineLocked(TR_Vlog_JITServer,
+            "ERROR: Found class loader but not chain for first loaded class %.*s", RECORD_NAME(record)
```
Comment (mpirvu) [inline runtime/compiler/runtime/JITServerAOTDeserializer.cpp:395]: Isn't this a normal case when SCC is not used? If so, we should not be generating error messages.

File: runtime/compiler/env/ClassLoaderTable.cpp (hunk 2 above)
Comment (mpirvu) [reply runtime/compiler/env/ClassLoaderTable.cpp:228]: Looking at the code, the "remove" function only detaches an entry from the bucket, but does not delete anything.
I don't remember if there was any technical issue with freeing the memory, but in the future we may want to actually free the memory, including the new name string that this PR has added. Please open a separate issue for it.

**Review by mpirvu — COMMENTED** [review]
Looks good overall. I have some small comments inline.

File: runtime/compiler/runtime/JITServerAOTDeserializer.cpp (hunk 3 above)
Comment (cjjdespres) [reply runtime/compiler/runtime/JITServerAOTDeserializer.cpp:395]: Yes, once the SCC isn't required we won't be looking up chains at all, so this message will be removed then. Right now the deserializer must be able to get the class chain (really the class chain offset) associated to a class loader in order to function properly, so to make this a standalone PR I had to add the extra check.

File: runtime/compiler/env/ClassLoaderTable.cpp (hunk 1 above)
Comment (cjjdespres) [reply runtime/compiler/env/ClassLoaderTable.cpp:210]: Yes, there's not much point to doing this without the upcoming JITServer AOT cache compilation changes.

File: runtime/compiler/env/ClassLoaderTable.cpp [hunk 4]
```diff
@@ -58,9 +58,25 @@ struct TR_ClassLoaderInfo
    TR_ClassLoaderInfo *_chainTableNext;
 #if defined(J9VM_OPT_JITSERVER)
    TR_ClassLoaderInfo *_nameTableNext;
+   // The ROM class name with either be in the local SCC or will be
```
[the remark this answers is not in the document]
Comment (cjjdespres) [reply runtime/compiler/env/ClassLoaderTable.cpp]: Fixed

File: runtime/compiler/env/ClassLoaderTable.cpp (hunk 2 above)
Comment (cjjdespres) [reply runtime/compiler/env/ClassLoaderTable.cpp:228]: I should probably check that the name string isn't NULL before freeing it, though.

Comment (cjjdespres) [reply runtime/compiler/env/ClassLoaderTable.cpp:228]: Oh, never mind. That shouldn't be necessary. I forgot that we now track names exactly when `getJITServerUseAOTCache()` is true, and I don't think that will ever switch from false to true during the operation of the class loader table.

**Comment by mpirvu** [comment]
This PR needs a rebase.

File: runtime/compiler/env/ClassLoaderTable.cpp [hunk 5]
```diff
@@ -183,28 +204,48 @@ TR_PersistentClassLoaderTable::associateClassLoaderWithClass(J9VMThread *vmThrea
...
+         {
+         chain = _sharedCache->pointerFromOffsetInSharedCache(chainOffset);
+         TR_ASSERT(_sharedCache->isPointerInSharedCache(chain), "Class chain must be in SCC");
+         }
+      }
+
+   // If we are using the JITServer AOT cache, we need to remember the name of clazz with or without chain.
+   // If we aren't using it, there is no point in continuing.
+   if (!chain && !useAOTCache)
       return;
```
Comment (mpirvu) [inline runtime/compiler/env/ClassLoaderTable.cpp:202]: Don't we miss the return here?

Comment (cjjdespres) [reply runtime/compiler/env/ClassLoaderTable.cpp:202]: I think it's there, in line 237 just below the diff?

https://github.com/eclipse-openj9/openj9/pull/18839/files#diff-f8ebc6246869c77a730d272b736cc0a8f2778232b1ce5ebc8eb4e1776db91194R237

Comment (cjjdespres) [reply runtime/compiler/env/ClassLoaderTable.cpp:202]: Based on our discussion I added early returns to the `if (_sharedCache)` block so we don't bother continuing if an SCC isn't present or we can't find the chain offset. These will be removed when in the main JITServer AOT cache PR.

**Review by mpirvu — APPROVED** [review]
LGTM

**Comment by mpirvu** [comment]
jenkins test sanity plinuxjit,xlinuxjit,zlinuxjit,alinux64jit,zlinux jdk17

**Comment by cjjdespres** [comment]
The failure on s390x was in `testJITServer_2`, with this message:

```
Failed to properly start server, it terminated prematurely with exit value: 1
Dumping the contents of log file: /home/jenkins/workspace/Test_openjdk17_j9_sanity.functional_s390x_linux_Personal_testList_0/aqa-tests/TKG/output_17068212305532/testJITServer_2/testServerComesUpAfterClient.server.out
////////////////////////////////////////////////////////////////////////
//// JITServer is currently a technology preview. Its use is not yet supported.
//// can't bind server address: Address already in use
////////////////////////////////////////////////////////////////////////
```

which is known to happen occasionally. (I haven't had time to implement any of what I talked about in https://github.com/eclipse-openj9/openj9/issues/14706#issuecomment-1883459860).

**Comment by mpirvu** [comment]
One of the JITServer tests failed on zLinux.

**Comment by mpirvu** [comment]
jenkins test sanity zlinuxjit, zlinux jdk21

**Comment by mpirvu** [comment]
jenkins test sanity zlinuxjit, zlinux jdk21

**Comment by mpirvu** [comment]
jenkins test sanity zlinuxjit,zlinux jdk21

**Comment by cjjdespres** [comment]
The failure in sanity zlinuxjit jdk21 is https://github.com/eclipse-openj9/openj9/issues/17474,

```
17:30:43   [ERR] Assertion failed at /home/jenkins/workspace/Build_JDK21_s390x_linux_jit_Personal/openj9/runtime/compiler/env/JITServerPersistentCHTable.cpp:172: classInfo
17:30:43   [ERR] 	subclass info cannot be null: ensure subclasses are loaded before superclass
```
The failure in sanity zlinuxjit jdk17 is this:

```
04:45:05  Testing: Create CRIU checkpoint image and restore once - testGetProcessRestoreStartTime
04:45:05  Test start time: 2024/02/02 04:45:04 Eastern Standard Time
04:45:05  Running command: bash /home/jenkins/workspace/Test_openjdk17_j9_sanity.functional_s390x_linux_jit_Personal_testList_1/aqa-tests/TKG/../../jvmtest/functional/cmdLineTests/criu/criuScript.sh /home/jenkins/workspace/Test_openjdk17_j9_sanity.functional_s390x_linux_jit_Personal_testList_1/aqa-tests/TKG/../../jvmtest/functional/cmdLineTests/criu /home/jenkins/workspace/Test_openjdk17_j9_sanity.functional_s390x_linux_jit_Personal_testList_1/jdkbinary/j2sdk-image/bin/java "-XX:+UseJITServer -Xjit:count=0 -XX:+JVMPortableRestoreMode  -Xtrace:print={j9vm.684-696,j9vm.699,j9vm.717-743,j9vm.746-747} --add-exports java.base/openj9.internal.criu=ALL-UNNAMED" org.openj9.criu.TimeChangeTest testGetProcessRestoreStartTime 1 false false
04:45:05  Time spent starting: 10 milliseconds
04:45:40  Time spent executing: 34884 milliseconds
04:45:40  Test result: FAILED
04:45:40  Output from test:
04:45:40   [OUT] start running script
04:45:40   [OUT] export GLIBC_TUNABLES=glibc.cpu.hwcaps=-XSAVEC,-XSAVE,-AVX2,-ERMS,-AVX,-AVX_Fast_Unaligned_Load
04:45:40   [OUT] export LD_BIND_NOT=on
04:45:40   [OUT] /home/jenkins/workspace/Test_openjdk17_j9_sanity.functional_s390x_linux_jit_Personal_testList_1/jdkbinary/j2sdk-image/bin/java -XX:+EnableCRIUSupport -XX:+UseJITServer -Xjit:count=0 -XX:+JVMPortableRestoreMode  -Xtrace:print={j9vm.684-696,j9vm.699,j9vm.717-743,j9vm.746-747} --add-exports java.base/openj9.internal.criu=ALL-UNNAMED -cp /home/jenkins/workspace/Test_openjdk17_j9_sanity.functional_s390x_linux_jit_Personal_testList_1/aqa-tests/TKG/../../jvmtest/functional/cmdLineTests/criu/criu.jar org.openj9.criu.TimeChangeTest testGetProcessRestoreStartTime 1
04:45:40   [OUT] Unhandled exception
04:45:40   [OUT] Type=Segmentation error vmState=0x00000000
04:45:40   [OUT] J9Generic_Signal_Number=00000018 Signal_Number=0000000b Error_Value=000000a9 Signal_Code=00000001
04:45:40   [OUT] Handler1=000003FFA8C49720 Handler2=000003FFA8B31A10 InaccessibleAddress=0000000000000000
04:45:40   [OUT] gpr0=000003FF1000C0F8 gpr1=000003FFA93A5F30 gpr2=000003FF1000C0F8 gpr3=0000000000000000
04:45:40   [OUT] gpr4=0000000000000200 gpr5=0000000000039088 gpr6=000003FFA86B82F8 gpr7=308317BD6FFFF1C0
04:45:40   [OUT] gpr8=000003FF1000C0F8 gpr9=000003FF00000001 gpr10=0000000000000000 gpr11=000003FF1000C050
04:45:40   [OUT] gpr12=000003FF10000A28 gpr13=00000000001FB468 gpr14=000003FFA8690790 gpr15=000003FFA897D600
04:45:40   [OUT] psw=000003FFA93A5F42 mask=0705200180000000 fpc=0008fe00 bea=000003FFA868329C
04:45:40   [OUT] fpr0 42aabc94c0000000 (f: 3221225472.000000, d: 1.469863e+13)
04:45:40   [OUT] fpr1 3e63a62300000000 (f: 0.000000, d: 3.659906e-08)
04:45:40   [OUT] fpr2 3e3ab28300000000 (f: 0.000000, d: 6.215952e-09)
04:45:40   [OUT] fpr3 3cc835f600000000 (f: 0.000000, d: 6.719843e-16)
04:45:40   [OUT] fpr4 3eb5480c0011ff00 (f: 1179392.000000, d: 1.268472e-06)
04:45:40   [OUT] fpr5 3e92576300000000 (f: 0.000000, d: 2.733075e-07)
04:45:40   [OUT] fpr6 3eccde3510000a28 (f: 268438048.000000, d: 3.441333e-06)
04:45:40   [OUT] fpr7 3e3a4a5100000000 (f: 0.000000, d: 6.121187e-09)
04:45:40   [OUT] fpr8 0000018d69352a82 (f: 1765091968.000000, d: 8.433044e-312)
04:45:40   [OUT] fpr9 000000000014ff48 (f: 1376072.000000, d: 6.798699e-318)
04:45:40   [OUT] fpr10 000003ffa40138b0 (f: 2751543552.000000, d: 2.172161e-311)
04:45:40   [OUT] fpr11 0000000000000000 (f: 0.000000, d: 0.000000e+00)
04:45:40   [OUT] fpr12 00061062f778fb95 (f: 4151901184.000000, d: 8.433044e-309)
04:45:40   [OUT] fpr13 000003ffb07dfaa8 (f: 2961046272.000000, d: 2.172265e-311)
04:45:40   [OUT] fpr14 000003fffee751b4 (f: 4276572672.000000, d: 2.172915e-311)
04:45:40   [OUT] fpr15 000003ffb07e0bd8 (f: 2961050624.000000, d: 2.172265e-311)
04:45:40   [OUT] Module=/lib64/libc.so.6
04:45:40   [OUT] Module_base_address=000003FFA9300000
04:45:40   [OUT] Target=2_90_20240201_121 (Linux 3.10.0-1160.83.1.el7.s390x)
04:45:40   [OUT] CPU=s390x (4 logical CPUs) (0x1ec1b1000 RAM)
04:45:40   [OUT] ----------- Stack Backtrace -----------
04:45:40   [OUT] __strncpy_vx+0x12 (0x000003FFA93A5F42 [libc.so.6+0xa5f42])
04:45:40   [OUT] getTrcBuf+0x370 (0x000003FFA8690790 [libj9trc29.so+0x10790])
04:45:40   [OUT] traceV+0xb54 (0x000003FFA8691A0C [libj9trc29.so+0x11a0c])
04:45:40   [OUT] doTracePoint+0xa34 (0x000003FFA8693484 [libj9trc29.so+0x13484])
04:45:40   [OUT] omrTrace+0x72 (0x000003FFA8693972 [libj9trc29.so+0x13972])
04:45:40   [OUT] internalFindClassInModule+0xe1e (0x000003FFA8C2E656 [libj9vm29.so+0x2e656])
04:45:40   [OUT] internalFindClassUTF8+0x2c (0x000003FFA8C31334 [libj9vm29.so+0x31334])
04:45:40   [OUT] resolveClassRef+0x23c (0x000003FFA8C89844 [libj9vm29.so+0x89844])
04:45:40   [OUT] resolveStaticFieldRefInto+0x100 (0x000003FFA8C8A4C8 [libj9vm29.so+0x8a4c8])
04:45:40   [OUT] resolveStaticFieldRef+0x34 (0x000003FFA8C8AA9C [libj9vm29.so+0x8aa9c])
04:45:40   [OUT] _ZN36VM_CRIUBytecodeInterpreterCompressed3runEP10J9VMThread+0x1dee8 (0x000003FFA8DCB168 [libj9vm29.so+0x1cb168])
04:45:40   [OUT] criuBytecodeLoopCompressed+0xee (0x000003FFA8DAD26E [libj9vm29.so+0x1ad26e])
04:45:40   [OUT] c_cInterpreter+0x64 (0x000003FFA8E12CC4 [libj9vm29.so+0x212cc4])
04:45:40   [OUT] ---------------------------------------
04:45:40   [OUT] JVMDUMP039I Processing dump event "gpf", detail "" at 2024/02/02 04:45:08 - please wait.
04:45:40   [OUT] JVMDUMP032I JVM requested System dump using '/home/jenkins/workspace/Test_openjdk17_j9_sanity.functional_s390x_linux_jit_Personal_testList_1/aqa-tests/TKG/output_17068631619043/cmdLineTester_criu_nonPortableRestore_3/core.20240202.044508.36729.0001.dmp' in response to an event
04:45:40   [OUT] JVMDUMP010I System dump written to /home/jenkins/workspace/Test_openjdk17_j9_sanity.functional_s390x_linux_jit_Personal_testList_1/aqa-tests/TKG/output_17068631619043/cmdLineTester_criu_nonPortableRestore_3/core.20240202.044508.36729.0001.dmp
04:45:40   [OUT] JVMDUMP032I JVM requested Java dump using '/home/jenkins/workspace/Test_openjdk17_j9_sanity.functional_s390x_linux_jit_Personal_testList_1/aqa-tests/TKG/output_17068631619043/cmdLineTester_criu_nonPortableRestore_3/javacore.20240202.044508.36729.0002.txt' in response to an event
04:45:40   [OUT] JVMDUMP010I Java dump written to /home/jenkins/workspace/Test_openjdk17_j9_sanity.functional_s390x_linux_jit_Personal_testList_1/aqa-tests/TKG/output_17068631619043/cmdLineTester_criu_nonPortableRestore_3/javacore.20240202.044508.36729.0002.txt
04:45:40   [OUT] JVMDUMP032I JVM requested Snap dump using '/home/jenkins/workspace/Test_openjdk17_j9_sanity.functional_s390x_linux_jit_Personal_testList_1/aqa-tests/TKG/output_17068631619043/cmdLineTester_criu_nonPortableRestore_3/Snap.20240202.044508.36729.0003.trc' in response to an event
04:45:40   [OUT] JVMDUMP010I Snap dump written to /home/jenkins/workspace/Test_openjdk17_j9_sanity.functional_s390x_linux_jit_Personal_testList_1/aqa-tests/TKG/output_17068631619043/cmdLineTester_criu_nonPortableRestore_3/Snap.20240202.044508.36729.0003.trc
04:45:40   [OUT] JVMDUMP032I JVM requested JIT dump using '/home/jenkins/workspace/Test_openjdk17_j9_sanity.functional_s390x_linux_jit_Personal_testList_1/aqa-tests/TKG/output_17068631619043/cmdLineTester_criu_nonPortableRestore_3/jitdump.20240202.044508.36729.0004.dmp' in response to an event
04:45:40   [OUT] JVMDUMP051I JIT dump occurred in 'Attach API initializer' thread 0x00000000001FB400
04:45:40   [OUT] JVMDUMP010I JIT dump written to /home/jenkins/workspace/Test_openjdk17_j9_sanity.functional_s390x_linux_jit_Personal_testList_1/aqa-tests/TKG/output_17068631619043/cmdLineTester_criu_nonPortableRestore_3/jitdump.20240202.044508.36729.0004.dmp
04:45:40   [OUT] JVMDUMP013I Processed dump event "gpf", detail "".
04:45:40   [OUT]  36687: Error (criu/files-reg.c:1702): File home/jenkins/workspace/Test_openjdk17_j9_sanity.functional_s390x_linux_jit_Personal_testList_1/aqa-tests/TKG/output_17068631619043/cmdLineTester_criu_nonPortableRestore_3/testOutput has bad size 5099 (expect 1341)
04:45:40   [OUT]  36687: Error (criu/files.c:1196): Unable to open fd=1 id=0x33
04:45:40   [OUT] Error (criu/cr-restore.c:1414): 36687 exited, status=1
04:45:40   [OUT] Error (criu/cr-restore.c:2293): Restoring FAILED.
04:45:40   [OUT] Removed test output files
04:45:40   [OUT] finished script
04:45:40  >> Success condition was not found: [Output match: Killed]
04:45:40  >> Success condition was not found: [Output match: PASSED: InternalCRIUSupport.getProcessRestoreStartTime()]
04:45:40  >> Failure condition was not found: [Output match: CRIU is not enabled]
04:45:40  >> Failure condition was not found: [Output match: Operation not permitted]
04:45:40  >> Failure condition was not found: [Output match: FAILED: InternalCRIUSupport.getProcessRestoreStartTime()]
04:45:40  >> Success condition was not found: [Output match: Thread pid mismatch]
04:45:40  >> Success condition was not found: [Output match: do not match expected]
04:45:40  >> Success condition was not found: [Output match: Unable to create a thread:]
04:45:40  >> Failure condition was not found: [Output match: Could not dump the JVM processes, err=-70]
04:45:40  >> Failure condition was not found: [Output match: User requested Java dump using]
```

I haven't seen this one before.

The failure in zlinux jdk17 is

```
18:08:57  ////////////////////////////////////////////////////////////////////////
18:08:57  //// JITServer is currently a technology preview. Its use is not yet supported.
18:08:57  //// can't bind server address: Address already in use
18:08:57  ////////////////////////////////////////////////////////////////////////
```

which is known. Strange that it's happening this often, though.

**Comment by mpirvu** [comment]
jenkins test sanity zlinux jdk17

**Comment by mpirvu** [comment]
zlinux jit failed again with
```
04:45:05  Testing: Create CRIU checkpoint image and restore once - testGetProcessRestoreStartTime
04:45:05  Test start time: 2024/02/02 04:45:04 Eastern Standard Time
04:45:05  Running command: bash /home/jenkins/workspace/Test_openjdk17_j9_sanity.functional_s390x_linux_jit_Personal_testList_1/aqa-tests/TKG/../../jvmtest/functional/cmdLineTests/criu/criuScript.sh /home/jenkins/workspace/Test_openjdk17_j9_sanity.functional_s390x_linux_jit_Personal_testList_1/aqa-tests/TKG/../../jvmtest/functional/cmdLineTests/criu /home/jenkins/workspace/Test_openjdk17_j9_sanity.functional_s390x_linux_jit_Personal_testList_1/jdkbinary/j2sdk-image/bin/java "-XX:+UseJITServer -Xjit:count=0 -XX:+JVMPortableRestoreMode  -Xtrace:print={j9vm.684-696,j9vm.699,j9vm.717-743,j9vm.746-747} --add-exports java.base/openj9.internal.criu=ALL-UNNAMED" org.openj9.criu.TimeChangeTest testGetProcessRestoreStartTime 1 false false
04:45:05  Time spent starting: 10 milliseconds
04:45:40  Time spent executing: 34884 milliseconds
04:45:40  Test result: FAILED
04:45:40  Output from test:
04:45:40   [OUT] start running script
04:45:40   [OUT] export GLIBC_TUNABLES=glibc.cpu.hwcaps=-XSAVEC,-XSAVE,-AVX2,-ERMS,-AVX,-AVX_Fast_Unaligned_Load
04:45:40   [OUT] export LD_BIND_NOT=on
04:45:40   [OUT] /home/jenkins/workspace/Test_openjdk17_j9_sanity.functional_s390x_linux_jit_Personal_testList_1/jdkbinary/j2sdk-image/bin/java -XX:+EnableCRIUSupport -XX:+UseJITServer -Xjit:count=0 -XX:+JVMPortableRestoreMode  -Xtrace:print={j9vm.684-696,j9vm.699,j9vm.717-743,j9vm.746-747} --add-exports java.base/openj9.internal.criu=ALL-UNNAMED -cp /home/jenkins/workspace/Test_openjdk17_j9_sanity.functional_s390x_linux_jit_Personal_testList_1/aqa-tests/TKG/../../jvmtest/functional/cmdLineTests/criu/criu.jar org.openj9.criu.TimeChangeTest testGetProcessRestoreStartTime 1
04:45:40   [OUT] Unhandled exception
04:45:40   [OUT] Type=Segmentation error vmState=0x00000000
04:45:40   [OUT] J9Generic_Signal_Number=00000018 Signal_Number=0000000b Error_Value=000000a9 Signal_Code=00000001
04:45:40   [OUT] Handler1=000003FFA8C49720 Handler2=000003FFA8B31A10 InaccessibleAddress=0000000000000000
04:45:40   [OUT] gpr0=000003FF1000C0F8 gpr1=000003FFA93A5F30 gpr2=000003FF1000C0F8 gpr3=0000000000000000
04:45:40   [OUT] gpr4=0000000000000200 gpr5=0000000000039088 gpr6=000003FFA86B82F8 gpr7=308317BD6FFFF1C0
04:45:40   [OUT] gpr8=000003FF1000C0F8 gpr9=000003FF00000001 gpr10=0000000000000000 gpr11=000003FF1000C050
04:45:40   [OUT] gpr12=000003FF10000A28 gpr13=00000000001FB468 gpr14=000003FFA8690790 gpr15=000003FFA897D600
04:45:40   [OUT] psw=000003FFA93A5F42 mask=0705200180000000 fpc=0008fe00 bea=000003FFA868329C
04:45:40   [OUT] fpr0 42aabc94c0000000 (f: 3221225472.000000, d: 1.469863e+13)
04:45:40   [OUT] fpr1 3e63a62300000000 (f: 0.000000, d: 3.659906e-08)
04:45:40   [OUT] fpr2 3e3ab28300000000 (f: 0.000000, d: 6.215952e-09)
04:45:40   [OUT] fpr3 3cc835f600000000 (f: 0.000000, d: 6.719843e-16)
04:45:40   [OUT] fpr4 3eb5480c0011ff00 (f: 1179392.000000, d: 1.268472e-06)
04:45:40   [OUT] fpr5 3e92576300000000 (f: 0.000000, d: 2.733075e-07)
04:45:40   [OUT] fpr6 3eccde3510000a28 (f: 268438048.000000, d: 3.441333e-06)
04:45:40   [OUT] fpr7 3e3a4a5100000000 (f: 0.000000, d: 6.121187e-09)
04:45:40   [OUT] fpr8 0000018d69352a82 (f: 1765091968.000000, d: 8.433044e-312)
04:45:40   [OUT] fpr9 000000000014ff48 (f: 1376072.000000, d: 6.798699e-318)
04:45:40   [OUT] fpr10 000003ffa40138b0 (f: 2751543552.000000, d: 2.172161e-311)
04:45:40   [OUT] fpr11 0000000000000000 (f: 0.000000, d: 0.000000e+00)
04:45:40   [OUT] fpr12 00061062f778fb95 (f: 4151901184.000000, d: 8.433044e-309)
04:45:40   [OUT] fpr13 000003ffb07dfaa8 (f: 2961046272.000000, d: 2.172265e-311)
04:45:40   [OUT] fpr14 000003fffee751b4 (f: 4276572672.000000, d: 2.172915e-311)
04:45:40   [OUT] fpr15 000003ffb07e0bd8 (f: 2961050624.000000, d: 2.172265e-311)
04:45:40   [OUT] Module=/lib64/libc.so.6
04:45:40   [OUT] Module_base_address=000003FFA9300000
04:45:40   [OUT] Target=2_90_20240201_121 (Linux 3.10.0-1160.83.1.el7.s390x)
04:45:40   [OUT] CPU=s390x (4 logical CPUs) (0x1ec1b1000 RAM)
04:45:40   [OUT] ----------- Stack Backtrace -----------
04:45:40   [OUT] __strncpy_vx+0x12 (0x000003FFA93A5F42 [libc.so.6+0xa5f42])
04:45:40   [OUT] getTrcBuf+0x370 (0x000003FFA8690790 [libj9trc29.so+0x10790])
04:45:40   [OUT] traceV+0xb54 (0x000003FFA8691A0C [libj9trc29.so+0x11a0c])
04:45:40   [OUT] doTracePoint+0xa34 (0x000003FFA8693484 [libj9trc29.so+0x13484])
04:45:40   [OUT] omrTrace+0x72 (0x000003FFA8693972 [libj9trc29.so+0x13972])
04:45:40   [OUT] internalFindClassInModule+0xe1e (0x000003FFA8C2E656 [libj9vm29.so+0x2e656])
04:45:40   [OUT] internalFindClassUTF8+0x2c (0x000003FFA8C31334 [libj9vm29.so+0x31334])
04:45:40   [OUT] resolveClassRef+0x23c (0x000003FFA8C89844 [libj9vm29.so+0x89844])
04:45:40   [OUT] resolveStaticFieldRefInto+0x100 (0x000003FFA8C8A4C8 [libj9vm29.so+0x8a4c8])
04:45:40   [OUT] resolveStaticFieldRef+0x34 (0x000003FFA8C8AA9C [libj9vm29.so+0x8aa9c])
04:45:40   [OUT] _ZN36VM_CRIUBytecodeInterpreterCompressed3runEP10J9VMThread+0x1dee8 (0x000003FFA8DCB168 [libj9vm29.so+0x1cb168])
04:45:40   [OUT] criuBytecodeLoopCompressed+0xee (0x000003FFA8DAD26E [libj9vm29.so+0x1ad26e])
04:45:40   [OUT] c_cInterpreter+0x64 (0x000003FFA8E12CC4 [libj9vm29.so+0x212cc4])
04:45:40   [OUT] ---------------------------------------
04:45:40   [OUT] JVMDUMP039I Processing dump event "gpf", detail "" at 2024/02/02 04:45:08 - please wait.
04:45:40   [OUT] JVMDUMP032I JVM requested System dump using '/home/jenkins/workspace/Test_openjdk17_j9_sanity.functional_s390x_linux_jit_Personal_testList_1/aqa-tests/TKG/output_17068631619043/cmdLineTester_criu_nonPortableRestore_3/core.20240202.044508.36729.0001.dmp' in response to an event
04:45:40   [OUT] JVMDUMP010I System dump written to /home/jenkins/workspace/Test_openjdk17_j9_sanity.functional_s390x_linux_jit_Personal_testList_1/aqa-tests/TKG/output_17068631619043/cmdLineTester_criu_nonPortableRestore_3/core.20240202.044508.36729.0001.dmp
04:45:40   [OUT] JVMDUMP032I JVM requested Java dump using '/home/jenkins/workspace/Test_openjdk17_j9_sanity.functional_s390x_linux_jit_Personal_testList_1/aqa-tests/TKG/output_17068631619043/cmdLineTester_criu_nonPortableRestore_3/javacore.20240202.044508.36729.0002.txt' in response to an event
04:45:40   [OUT] JVMDUMP010I Java dump written to /home/jenkins/workspace/Test_openjdk17_j9_sanity.functional_s390x_linux_jit_Personal_testList_1/aqa-tests/TKG/output_17068631619043/cmdLineTester_criu_nonPortableRestore_3/javacore.20240202.044508.36729.0002.txt
04:45:40   [OUT] JVMDUMP032I JVM requested Snap dump using '/home/jenkins/workspace/Test_openjdk17_j9_sanity.functional_s390x_linux_jit_Personal_testList_1/aqa-tests/TKG/output_17068631619043/cmdLineTester_criu_nonPortableRestore_3/Snap.20240202.044508.36729.0003.trc' in response to an event
04:45:40   [OUT] JVMDUMP010I Snap dump written to /home/jenkins/workspace/Test_openjdk17_j9_sanity.functional_s390x_linux_jit_Personal_testList_1/aqa-tests/TKG/output_17068631619043/cmdLineTester_criu_nonPortableRestore_3/Snap.20240202.044508.36729.0003.trc
04:45:40   [OUT] JVMDUMP032I JVM requested JIT dump using '/home/jenkins/workspace/Test_openjdk17_j9_sanity.functional_s390x_linux_jit_Personal_testList_1/aqa-tests/TKG/output_17068631619043/cmdLineTester_criu_nonPortableRestore_3/jitdump.20240202.044508.36729.0004.dmp' in response to an event
04:45:40   [OUT] JVMDUMP051I JIT dump occurred in 'Attach API initializer' thread 0x00000000001FB400
04:45:40   [OUT] JVMDUMP010I JIT dump written to /home/jenkins/workspace/Test_openjdk17_j9_sanity.functional_s390x_linux_jit_Personal_testList_1/aqa-tests/TKG/output_17068631619043/cmdLineTester_criu_nonPortableRestore_3/jitdump.20240202.044508.36729.0004.dmp
04:45:40   [OUT] JVMDUMP013I Processed dump event "gpf", detail "".
04:45:40   [OUT]  36687: Error (criu/files-reg.c:1702): File home/jenkins/workspace/Test_openjdk17_j9_sanity.functional_s390x_linux_jit_Personal_testList_1/aqa-tests/TKG/output_17068631619043/cmdLineTester_criu_nonPortableRestore_3/testOutput has bad size 5099 (expect 1341)
04:45:40   [OUT]  36687: Error (criu/files.c:1196): Unable to open fd=1 id=0x33
04:45:40   [OUT] Error (criu/cr-restore.c:1414): 36687 exited, status=1
```

**Comment by cjjdespres** [comment]
I think this component of the test was added a few weeks ago, in https://github.com/eclipse-openj9/openj9/pull/18680.

**Comment by mpirvu** [comment]
Merging based on the comments above regrading the CRIU failures.

**Review by AlexeyKhrabrov — COMMENTED** [review]
I finally had a chance to look at some of the "SCC-less AOT cache" PRs including this one. I have a couple of inline comments about handling memory allocation failures.

**Pull request merged**